contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -2324,6 +2324,14 @@ modify_insert_stmt(InsertStmt *stmt, Oid relid)
 	HeapTuple	tuple;
 	List	   *insert_col_list = NIL,
 			   *temp_col_list;
+	char		relkind = get_rel_relkind(relid);
+
+	if(relkind == RELKIND_VIEW || relkind == RELKIND_MATVIEW)
+	{
+		ereport(ERROR,
```

Comment:
We should map this error code to TSQL error code.

Reply:
Done